config.go

```diff
@@ -106,6 +108,27 @@ func (c *Config) Discover() error {
 		}
 	}
 
+	// Finally, we'll fill in any internal plugins that haven't already been
+	// discovered on disk.
```

Comment:
I took a brief look at go-plugin but it doesn't seem easy to add version in a backwards-compatible way so I'm going to defer this. It may be possible to do something like pass an ENV alongside the magic cookie but I think that leaves enforcement of the version constraint up to the plugin instead of the main binary, which seems problematic.